app/models/user.rb

```diff
@@ -143,6 +142,14 @@ def course_instructor?
     @course_instructor ||= courses_users.exists?(role: CoursesUsers::Roles::INSTRUCTOR_ROLE)
   end
 
+  def active_course_instructor?
```

Comment:
This is a reasonable instance method, but I think it should be called directly from the controller rather than in the validation callback.